content/blog/pulumi-components/index.md

```diff
@@ -0,0 +1,339 @@
+    {
+        ["BucketName"] = secureBucket.BucketName
+    };
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	secure "github.com/myorg/secure-s3-component/sdk/go/secure-s3-component"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		tags := make(map[string]string)
+		tags["Environment"] = "production"
+		tags["Department"] = "engineering"
+		
+		secureBucket, err := secure.NewSecureBucket(ctx, "my-secure-bucket", &secure.SecureBucketArgs{
+			BucketName: pulumi.String("my-company-secure-assets"),
+			Tags:       pulumi.ToStringMap(tags),
+		})
+		if err != nil {
+			return err
+		}
+		
+		ctx.Export("bucketName", secureBucket.BucketName)
+		return nil
+	})
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language java %}}
+
+```java
+package mypackage;
+
+import com.pulumi.Pulumi;
+import com.pulumi.core.Output;
+import com.myorg.secures3component.SecureBucket;
+import com.myorg.secures3component.SecureBucketArgs;
+
+import java.util.Map;
+
+public class App {
+    public static void main(String[] args) {
+        Pulumi.run(ctx -> {
+            var tags = Map.of(
+                "Environment", "production",
+                "Department", "engineering"
+            );
+            
+            var bucketArgs = SecureBucketArgs.builder()
+                .bucketName("my-company-secure-assets")
+                .tags(tags)
+                .build();
+                
+            var secureBucket = new SecureBucket("my-secure-bucket", bucketArgs);
+            
+            ctx.export("bucketName", secureBucket.bucketName());
+        });
+    }
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+name: my-infrastructure
+runtime: yaml
+description: A stack that uses SecureBucket component
+
+packages:
+  secure-s3-component: github.com/myorg/secure-s3-component@v1.0.0
+  
+resources:
+  secureBucket:
+    type: secure-s3-component:SecureBucket
+    properties:
+      bucketName: my-company-secure-assets
+      tags:
+        Environment: production
+        Department: engineering
+
+outputs:
+  bucketName: ${secureBucket.bucketName}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+## Pro Tips
+
+- **Tagging Releases**: If you're working with Git-based components, you can tag version releases (both stable and preview versions). This ensures downstream users can easily choose specific versions without chasing a moving target.
+
+- **Managing Dependencies**: Once you've run `pulumi package add <repo-url>`, Pulumi will generate a local SDK in your project and add an entry to the `packages` section of your Pulumi.yaml file. Check in these files if you want fully reproducible builds; or add them to `.gitignore` if you prefer to regenerate them on each checkout. When using `.gitignore`, team members can simply run `pulumi install` to restore all packages listed in Pulumi.yaml and regenerate the SDK.
```

Comment:
> add an entry to the `packages` section of your Pulumi.yaml file

This was planned but hasn't shipped yet (have to be careful with how this interacts with parameterized provider users). We'll ship it as a fast follow-up.

```suggestion
- **Managing Dependencies**: Once you've added an entry to the packages section of your Pulumi.yaml file, you can run `pulumi install` to generate a local SDK in your project. This command will process all packages listed in your Pulumi.yaml and create the necessary SDK files. Check in these files if you want fully reproducible builds; or add them to `.gitignore` if you prefer to regenerate them on each checkout. When using `.gitignore`, team members will need to run `pulumi install` after checkout to regenerate the SDK.
```

Reply:
should we be mentioning it in the blog if it is a fast follow?

Reply:
My edit is truthful at launch.